### ISSUE TITLE
IPSec between Libreswan and Racoon using AH

### ISSUE THREAD

**Issue by Yams-github24** [libreswan/libreswan#1826]
Hi,

We are trying to configure IPSec connection between Libreswan and Racoon using AH with certificate based authentication.
With this SAs and SPs are getting created and able to send the packet from racoon to libreswan end and vice versa, but the packet is not received by the application. 

This is happening only with AH. with ESP the same is working fine.

Can you please let us know if there is any limitation with AH while using IPSec connection between Libreswan and Racoon?

Version used: Linux Libreswan 4.3 
Sample config:
        authby=rsasig
        ike=aes-sha512;dh14
        phase2=ah  <--- 
        ah=sha256-dh14  <--- 
        ikev2=no
        sha2-truncbug=yes
        aggressive=no

**Comment by cagney** [comment]
I'd try something like `esp=null-sha2-dh14`.  Using ESP with NULL encryption is preferable to AH.

For debugging this, ideal would be to reproduce this using Libreswan's test framework (the NetBSD VMs have raccoon installed) and mainline Libreswan.  But off hand I'd compare the routing and kernel state/policy for the two alternatives. 

BTW, I'm surprised that sha2-truncbug is needed.

**Comment by paulwouters** [comment]
whether or not you need sha2-truncbug really depends on the Android/Linux kernel version. You are best to avoid sha256 and use sha384 or sha512.

But also, yes use esp null instead of ah is good.

racoon btw, is abandoned and no longer considered secure by its own people. You should stop using it:

From https://ipsec-tools.sourceforge.net/

```The development of ipsec-tools has been ABANDONED.

ipsec-tools has security issues, and you should not use it. Please switch to a secure alternative!
```

Also, you should switch from IKEv1 to IKEv2, see RFC 9395: https://datatracker.ietf.org/doc/html/rfc9395

**Comment by Yams-github24** [comment]
Thank you for the quick response.

- with "esp=null-sha2-dh14",  IPSec connection is working between Racoon and Libreswan. 
- with esp and all supported algorithms, IPSec connection is going through.
- Facing issues only with AH between racoon and Libreswan. Even with sha256, sha512, sha384, connection is not established.

Considering the interoperability support between racoon and libreswan, can you suggest if there is any workaround or tunable parameters which supports the IPSec connectivity using AH?

**Comment by cagney** [comment]
Is there a reason for not using ESP with NULL encryption?  It's functionally equivalent to AH in tunnel mode and more likely to be around long term.

Please also remember that Libreswan 4.3 is very old.  Your best next step is to upgrade your local libreswan to 5.0 or mainline and then demonstrate the problem is still present.

**Comment by letoams** [comment]
On Thu, 3 Oct 2024, Yaminni wrote:

> Thank you for the quick response.
>
>  *  with "esp=null-sha2-dh14", IPSec connection is working between Racoon and Libreswan.
>  *  with esp and all supported algorithms, IPSec connection is going through.
>  *  Facing issues only with AH between racoon and Libreswan. Even with sha256, sha512, sha384, connection is not
>     established.

Is your connection passing through a NAT? That might not work as well
with AH?

> Considering the interoperability support between racoon and libreswan, can you suggest if there is any workaround or
> tunable parameters which supports the IPSec connectivity using AH?

racoon is dead and its authors tell you to migrate away from it. Without
logs we cannot tell you why it is failing. Without knowing the cause of
failure, we cannot make any further suggestions. But the RFCs clearly
say "do not use AH, use ESP-NULL instead".

They also tell you to migrate away from IKEv1 to IKEv2.

In other words, you are asking the wrong questions. Get that racoon
endpoint replaced with a modern IKEv2 supporting IPsec stack.

Paul
